### DISCUSSION TITLE
Parameterising a value (a function) for entire test file?

### DISCUSSION THREAD

**Discussion by venaturum** [pytest-dev/pytest#8778] [Q&A]
Hi I'm having a bit of trouble figuring out how to parameterise an entire test file, any help would be much appreciated.

I have a test*.py file containing:

- a fixture which utilises a function "my_func"
- tests which utilise a function "my_func", and the fixture

I have several different functions that this function "my_func" could be.  At the moment I'm duplicating my test file for each possible value of "my_func".  Is it possible to parameterise this in pytest and just run with the one file?

Here is a simplified example of the current setup.  

```
import pytest

@pytest.fixture
def my_fix():
    a = my_obj()
    a.method(my_func(x))
    return a

def test_1(my_fix):
    assert my_fix.method(y) == my_func(z)
```

Cheers.

**Comment by The-Compiler** [comment]
It sounds like you're looking for [parametrized fixtures](https://docs.pytest.org/en/6.2.x/fixture.html#parametrizing-fixtures).

**Comment by venaturum** [comment]
I have tried that and failed, but perhaps that is my own inexperience. I think the issue might be that whatever parameter the fixture is using, the test methods needs to be running with the same parameter value. I suppose I could just give up on using fixtures, parameterise the methods and just call a function which returns a value based on the parameter. I'd rather do it the proper way if such a thing exists though.

**Reply by RonnyPfannschmidt** [reply]
in that case, create a new function object that always passes the fixed value, and use that everywhere

**Reply by venaturum** [reply]
Thanks, got it sorted!
